Services/PaymentsService.cs

```diff
@@ -69,14 +66,21 @@ private Receipt SaveReceipt(Order order, PaymentType type, int? couponId)
                 _couponService.Update(coupon);
             }
 
-            var receipt = new Receipt() {
+            var loyaltyPointsUsed = Math.Min(order.LoyaltyPointsToUse, order.Customer.LoyaltyPoints);
+            loyaltyPointsUsed = Math.Min(loyaltyPointsUsed, Convert.ToInt32(totalPrice * 100));
+            totalPrice -= (float)loyaltyPointsUsed / 100;
```

Comment:
I'm not sure I understand this part 😅 could you pls explain how loyalty points are supposed to work? (in general, not this code particularly)

or atleast the 2nd line `loyaltyPointsUsed = Math.Min(loyaltyPointsUsed, Convert.ToInt32(totalPrice * 100));`, where does 2nd arg `Convert.ToInt32(totalPrice * 100)` supposed to mean?

Reply:
Each loylty point is 1 cent, lets say you have 1000 loyalty points and you specify to use them all but your order is only 1 euro. This line takes the price in cents and checks if you want to use more loyalty points than the price. so in this case 100 < 1000 so you will use 100 loyalty points and will have 900 left over.

Reply:
In general, for each euro of total price you get 1 point, each point is worth 1 cent, you can specify how many points you want to use and these mins just check if you specified more than you have or more than the price in cents.